source/utilities.cc

```diff
@@ -745,3 +745,65 @@ namespace aspect
     template std_cxx1x::array<double,3> spherical_coordinates<3>(const Point<3> &position);
   }
 }
+
+namespace dealii
+{
+  namespace Utilities
+  {
+    namespace MPI
+    {
+      template<int dim>
+      Tensor<1,dim> 
+      reduce_tensor(const Tensor<1,dim> &local, const MPI_Comm &mpi_communicator)
+      {
+        std::vector<double> entries(dim);
+        std::vector<double> global_entries(dim);
+        Tensor<1,dim> global;
+
+        for(unsigned int i=0; i<dim; ++i)
+          entries[i] = local[i];
+
+        MPI_Allreduce (&entries[0], &global_entries[0], dim, MPI_DOUBLE,
+                       MPI_SUM, mpi_communicator);
+
+        for(unsigned int i=0; i<dim; ++i)
+          global[i] = global_entries[i];
+
+        return global;
+      }
+
+      template<int dim>
+      SymmetricTensor<2,dim> 
+      reduce_symmetric_tensor(const SymmetricTensor<2,dim> &local, const MPI_Comm &mpi_communicator)
+      {
+        unsigned int n_entries = dim*(dim+1)/2;
```

Comment:
Otherwise looks reasonable -- thanks for moving this to deal.II

I don't think you need fallbacks here -- we've always made the assumption that the ASPECT dev versions build on deal.II dev versions, and made sure we have a deal.II release before every ASPECT release.